### ISSUE TITLE
Flash support

### ISSUE THREAD

**Issue by letalvoj** [meetfranz/plugins#61]
Hi, 

I tried to create a plugin for Spotify and the problem seems to be a lack of Flash support. I downloaded flash for chromium but I still get this message:
<img width="369" alt="screen shot 2016-10-14 at 14 29 24" src="https://cloud.githubusercontent.com/assets/5262321/19387656/e654627c-921b-11e6-9562-3338b3e67e15.png">

Thanks
V.

**Comment by ryan-voltaire** [comment]
I had the same issue when I was trying to create  a plug-in for Pandora. It seems we may be trying to achieve the same thing.
Let quickly say that I have no experience in JS at all, but I understand some of the basics of what is kind of letting these plug-ins run.

I believe in the package.jason file you need to do something with Webview Options. I looked at the Electron documentation on webview options that was linked in and saw a spot called Plugins. This is the code that was there.

`<webview src="https://www.github.com/" plugins></webview>`

It says that it enables web browser plugins. Maybe doing something with this will allow your Spotify plugin to have access to flash. I'm not really too sure, but hope this may lead somewhere. I would like to use Pandora :disappointed:

**Comment by discipleofcode** [comment]
Yup, I'm also in need of flash support. What I've found out is that you can add flash support in electron like this -> https://github.com/electron/electron/blob/master/docs/tutorial/using-pepper-flash-plugin.md

Unfortunately I don't have any idea how to invoke app.getPath('pepperFlashSystemPlugin') on plugin level. Any ideas?

**Comment by letalvoj** [comment]
So does any of the included plugins require Flash? If yes can we dig-out the sources from the bundled app?

**Comment by genbtc** [comment]
I have gotten flash to work in franz. i will do a blog post later.

**Comment by discipleofcode** [comment]
Great!

**Comment by genbtc** [comment]
I Have written the blogpost here. Hope it helps, its kinda fiddly. https://mrlithium.blogspot.com/2016/11/enabling-pepper-flash-for-spotify-in.html

**Comment by discipleofcode** [comment]
It's nice that it works but as we need to edit Franz files directly (not for example override it from plugin level) it won't be really usable for non-tech folks as it will probably break on next Franz update. There is still room for improvement, as pepperflash library is not open source we could just load it depending on system from their respective path if it exists. Well, good job, but we will need some attention from Franz developers for this issue. If they won't implement such feature (at least ability to load flash/other 3rd library from plugin level) I will look for some other platform.

**Comment by genbtc** [comment]
agreed

**Comment by ataraxus** [comment]
@letalvoj any updates? Could i lend you a hand? It would be awesome to have spotify in franz ;)

**Comment by letalvoj** [comment]
Nope, sorry. And Spotify guys already fixed the bug which caused the app to write GBs of data per day to local storage. Thus I have no incentive to make it work anyway.
